src/implementations/minecraft/mod.rs

```diff
@@ -37,28 +38,50 @@ use crate::traits::t_configurable::PathBuf;
 use crate::traits::t_server::State;
 use crate::traits::TInstance;
 use crate::types::{InstanceUuid, Snowflake};
-use crate::util::{download_file, format_byte, format_byte_download, unzip_file};
+use crate::util::{download_file, format_byte, format_byte_download, unzip_file, dont_spawn_terminal};
 
 use self::players_manager::PlayersManager;
-use self::util::{get_fabric_jar_url, get_paper_jar_url, get_jre_url, get_vanilla_jar_url, read_properties_from_path};
+use self::util::{get_jre_url, get_server_jar_url, read_properties_from_path};
 
-#[derive(Debug, Clone, Copy, TS, Serialize, Deserialize)]
+#[derive(Debug, Clone, TS, Serialize, Deserialize)]
 #[serde(rename = "MinecraftFlavour", rename_all = "snake_case")]
 #[ts(export)]
 pub enum Flavour {
     Vanilla,
-    Fabric,
-    Paper,
+    Fabric {
+        loader_version: Option<String>,
+        installer_version: Option<String>,
```

Comment:
Probably should use the [new type idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) to make sure we don't pass in random strings as a version